packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -627,16 +645,72 @@ export function allocateChildren(vnode: VCustomElement, vm: VM) {
             );
         }
     }
+
+    // If any of the children being allocated are VFragments, we remove the text delimiters and flatten all immediate
+    // children VFragments to avoid them interfering with default slot behavior.
+
+    // If any of the VFragments were not stable, the children are marked dynamic.
+    let allocatedChildren = children;
+    if (children.some((vnode) => vnode && isVFragment(vnode))) {
+        const { stable, flattenedChildren } = flattenFragmentChildren(children);
+        allocatedChildren = flattenedChildren;
+        vnode.children = flattenedChildren;
+
+        if (!stable) {
```

Comment:
According to the coverage test, this `if` block is always `true`. (I.e. `stable` is always false.) It would be great to add tests for the `else` case.

(FYI, how I ran the coverage tests:

```sh
rm -fr coverage && COVERAGE=1 yarn test && COVERAGE=1 DISABLE_SYNTHETIC=1 yarn test && yarn run coverage
```

)

Reply:
Do we have the ability to write unit tests for `engine-core`? (might be a stupid question but I don't see any).

Right now, VFragments implement a `stable` flag but no real feature uses it yet. I added this in just to make it future-proof, but now I'm thinking we might want to remove this and add it back in once we have a feature that actually uses it (and can test it).

Reply:
Ah, I didn't realize that. In that case, I'd just add a comment saying that it's unused but we know why. Future-proofing with a one-line `if` statement is a good idea IMO.

And no, we don't test `@lwc/engine-core` in isolation at all.

Reply:
Ah or alternatively, I guess if `stable` is always `true`, we can just avoid computing it entirely if you want. That would simplify the `flattenFragmentChildren` function. I'm on the fence about which approach is better.